### PR TITLE
OSDOCS-5851-Newer: Document post-install AWS LZ work

### PR DESCRIPTION
**Pull request by dfitzmau**
[OSDOCS-5851](https://issues.redhat.com/browse/OSDOCS-5851)

Version(s):
4.14

Link to docs preview:
* [AWS Local Zone tasks (post-install)](https://65877--docspreview.netlify.app/openshift-enterprise/latest/post_installation_configuration/aws-compute-edge-tasks)
* [Sample YAML for a compute machine set custom resource on AWS (Creating a compute machine set on AWS)](https://65877--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-aws#machineset-yaml-aws_creating-machineset-aws)
* [Sample YAML for a compute machine set custom resource on AWS (Creating infrastructure machine sets)](https://65877--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets#machineset-yaml-aws_creating-infrastructure-machinesets)

QE review:
- [ ] QE has approved this change.

Additional information:
* PR continues work from https://github.com/openshift/openshift-docs/pull/63729/files

### CODE REVIEW

File: post_installation_configuration/cluster-tasks.adoc
Comment (dfitzmau) [inline post_installation_configuration/cluster-tasks.adoc:551]: This removes the AWS work from the Post-installation configuration / Cluster tasks section. See https://docs.openshift.com/container-platform/4.13/post_installation_configuration/cluster-tasks.html#installation-extend-edge-nodes-aws-local-zones_post-install-cluster-tasks

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (dfitzmau) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:35]: How about I add this module as either an additional resource link or as a prerequisite in nw-cluster-mtu-change.adoc ?

File: modules/machineset-creating.adoc
Comment (dfitzmau) [inline modules/machineset-creating.adoc:208]: Need to fix as it does not match https://github.com/openshift/openshift-docs/pull/63729/files

**Comment by dfitzmau** [comment]
COpied content from https://github.com/openshift/openshift-docs/pull/63729

/label peer-review-needed

File: modules/installation-aws-add-local-zone-locations.adoc
Comment (skrthomas) [reply modules/installation-aws-add-local-zone-locations.adoc:16]: A complete sentence is needed before the colon. 
IBMSG: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-colons#introductory-text

File: modules/installation-aws-add-local-zone-locations.adoc
Comment (skrthomas) [inline modules/installation-aws-add-local-zone-locations.adoc:63]: Instead of making this a list-type item, can it be a callout in the code? So put a <1> on line 57 then list this info as a callout?

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (skrthomas) [inline modules/post-install-edge-aws-extend-machineset.adoc:9]:
```suggestion
After you create subnets in an AWS Local Zones, you can create a machine set manifest.
```

Comment (skrthomas) [reply modules/post-install-edge-aws-extend-machineset.adoc:9]: I was thinking, since you use "After" the past-tense "created" isn't necessary.

Comment (skrthomas) [reply modules/post-install-edge-aws-extend-machineset.adoc:9]: Should it be a prerequisite for this procedure that you must have subnets in an AWS Local Zone?

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (skrthomas) [inline modules/post-install-edge-aws-extend-machineset.adoc:21]:
```suggestion
* Manually preserve`edge` machine pool labels when creating the machine set manifest by gathering the AWS API and running the following command:
```

Comment (skrthomas) [reply modules/post-install-edge-aws-extend-machineset.adoc:21]: I suggest this, just to make the procedure step more actionable from the beginning and clarify "those" labels.

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (skrthomas) [inline modules/post-install-edge-aws-extend-machineset.adoc:38]: My same comment as above, I think these need to be code callouts instead of the current presentation. See: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output

File: modules/post-install-existing-local-zone-subnet.adoc
Comment (skrthomas) [inline modules/post-install-existing-local-zone-subnet.adoc:13]:
```suggestion
The following subsections include steps using CloudFormation templates. Considering the limitation of NAT Gateways in AWS Local Zones, CloudFormation templates support only public subnets. You can adapt it to your needs. Additionally, you can reuse the template to create public subnets for each edge location to where you need to extend your cluster.
```

Comment (skrthomas) [reply modules/post-install-existing-local-zone-subnet.adoc:13]: IMO, you might as well remove the duplicated statement that "you can adapt/modify the CloudFormation to meet your needs and just add that last sentence to the paragraph before. Just a suggestion!

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (skrthomas) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:17]:
```suggestion
* For more information about AWS Local Zones, the supported instances types, and services, see link:https://aws.amazon.com/about-aws/global-infrastructure/localzones/features/[AWS Local Zones features] in the AWS documentation 
```

**Review by skrthomas — COMMENTED** [review]
A few comments/suggestions for you! Nice work :)

**Comment by dfitzmau** [comment]
> A few comments/suggestions for you! Nice work :)

Thanks, @skrthomas , for the excellent review, as always :+1: ! I incorporated all of your feedback :heavy_check_mark:

**Comment by dfitzmau** [comment]
HI @mtulio . In the Creating a subnet in AWS Local Zones section, I see the following example output:

> arn:aws:cloudformation:us-east-1:123456789012:stack/<subnet_stack_name>/dbedae40-2fd3-11eb-820e-12a48460849f

Is "arn:" correct?

File: modules/installation-aws-add-local-zone-locations.adoc
Comment (dfitzmau) [inline modules/installation-aws-add-local-zone-locations.adoc:57]: Docs team check: Check callouts that targets commands that span multiple lines works on Customer Portal.

Comment (kalexand-rh) [reply modules/installation-aws-add-local-zone-locations.adoc:57]: You'll need to validate this after the content is merged.

File: modules/machineset-yaml-aws.adoc
Comment (mtulio) [inline modules/machineset-yaml-aws.adoc:125]: Maybe it will be better addressed as the following to be in sync with the section `Creating a subnet in AWS Local Zones`:
```suggestion
              id: <value_of_PublicSubnetIds>
```

File: modules/machineset-yaml-aws.adoc
Comment (mtulio) [inline modules/machineset-yaml-aws.adoc:18]: We'll need to add ifdef here for edge compute pool. The `us-east-1a` is not a valid Local Zone Name, furthermore, the "sample snippet" does not set the zone name in `availabilityZone`.

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (mtulio) [inline modules/post-install-edge-aws-extend-machineset.adoc:48]: In the section `Sample YAML for a compute machine set custom resource on AWS` we are showing `us-east-1-nyc-1a` . It may not be required, but to improve the readability I suggest to synchronize the examples:

```suggestion
.Example output for Local Zone `us-east-1-nyc-1a`
[source,terminal]
----
[
    {
        "ZoneName": "us-east-1-nyc-1a",
        "ParentZoneName": "us-east-1f",
        "GroupName": "us-east-1-nyc-1",
```

**Review by mtulio — CHANGES_REQUESTED** [review]
Hey @dfitzmau . A few suggestions, thanks for owning it. I will close the original PR https://github.com/openshift/openshift-docs/pull/63729 .

File: modules/machineset-yaml-aws.adoc
Comment (dfitzmau) [reply modules/machineset-yaml-aws.adoc:125]: Good catch. Updated!

File: modules/machineset-yaml-aws.adoc
Comment (dfitzmau) [reply modules/machineset-yaml-aws.adoc:19]: Thanks. I see the following in the snippet, so does this resolve your `availabilityZone` query?

Screenshot from 2023-10-17 09-43-49

File: modules/nw-cluster-mtu-change.adoc
Comment (stevsmit) [inline modules/nw-cluster-mtu-change.adoc:14]: Might wanna remove this extra space

File: modules/nw-cluster-mtu-change.adoc
Comment (stevsmit) [inline modules/nw-cluster-mtu-change.adoc:352]: Maybe we could condense this? or not. it's just a suggestion.

```suggestion
* Verify that the node in your cluster uses the MTU that you specified in the previous procedure by entering the following command:
```

File: modules/post-install-edge-aws-extend-cluster.adoc
Comment (stevsmit) [inline modules/post-install-edge-aws-extend-cluster.adoc:7]: There are two modules with this name? They don't conflict with one another, just pointing it out

Extend existing clusters to use AWS Local Zone

File: modules/post-install-edge-aws-extend-cluster.adoc
Comment (stevsmit) [inline modules/post-install-edge-aws-extend-cluster.adoc:13]: Because these are fragments, they do not need periods. https://www.ibm.com/docs/en/ibm-style?topic=punctuation-periods#lists

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (stevsmit) [inline modules/post-install-edge-aws-extend-machineset.adoc:7]: for an AWS Local Zones node?

```suggestion
= Creating a machine set manifest for an AWS Local Zones node
```

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (stevsmit) [inline modules/post-install-edge-aws-extend-machineset.adoc:17]: Do we maybe want to make this more concise? Also, typo.

```suggestion
The following procedure details how you can create a machine set configuration that matches the `edge` compute pool configuration.
```

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (stevsmit) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:9]: We rarely say "to meet your needs." I'll let you decide whether to keep it, but the comma isn't needed:

```suggestion
After installing {product-title} on Amazon Web Services (AWS), you can further configure AWS Local Zones and an edge compute pool so that you can expand and customize your cluster.
```

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (stevsmit) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:26]:
```suggestion
network so that your cluster infrastructure can support Local Zone subnets. 
```

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (stevsmit) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:58]: Not a big deal, but the friendly reference text isn't the same as the module if you want them to match.

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (stevsmit) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:66]: Same here

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (stevsmit) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:68]: _Technically_ links to "Controlling pod placement using node taints"

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (stevsmit) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:72]:
```suggestion
* Optional: Use the AWS Load Balancer (ALB) Operator to expose a pod from a targeted edge worker node to services that run inside of a Local Zone subnet from a public network.
```

**Comment by stevsmit** [comment]
@dfitzmau Some nits to consider

File: modules/machineset-creating.adoc
Comment (dfitzmau) [reply modules/machineset-creating.adoc:214]: Good point. HI @mtulio . Should I update this line for AWS LZ?

Comment (mtulio) [reply modules/machineset-creating.adoc:214]: `us-east-1a` is not a Local Zone, but it is valid in a cluster with Local Zones enabled.
I can't see anything to change here.

File: modules/installation-aws-add-local-zone-locations.adoc
Comment (mtulio) [reply modules/installation-aws-add-local-zone-locations.adoc]: @dfitzmau I think you need to keep the zone name here to be didactic to user: `us-east-1-nyc-1` != `us-east-1-nyc-1a`.
Even more, the zone group `us-east-1-nyc-1` could have one or more zones (currently there is one, but the documentation may not follow AWS updates), if you want to keep the suggestion you may want adding the plural: `US East (New York) zones`

Comment (dfitzmau) [reply modules/installation-aws-add-local-zone-locations.adoc]: Thanks, @mtulio . Updated!

File: modules/installation-aws-add-local-zone-locations.adoc
Comment (mtulio) [inline modules/installation-aws-add-local-zone-locations.adoc:60]: @dfitzmau still incorrect, the zone name has been added again with wrong value.

```suggestion
<1> For `<value_of_GroupName>`, specify the name of the group of the Local Zone where you want to create subnets. For example, specify `us-east-1-nyc-1` to use the zone `us-east-1-nyc-1a` (US East New York).
```

**Review by mtulio — CHANGES_REQUESTED** [review]
Hi, one typo. Thanks

File: modules/installation-aws-add-local-zone-locations.adoc
Comment (mtulio) [reply modules/installation-aws-add-local-zone-locations.adoc:60]: cc @kalexand-rh (@dfitzmau mentioned he'll be on PTO the next two days)

File: modules/machineset-yaml-aws.adoc
Comment (yunjiang29) [reply modules/machineset-yaml-aws.adoc:55]: for [awsproviderconfig.openshift.io/v1beta1](https://github.com/openshift/openshift-docs/pull/65877/files#diff-099fb946e77b9a692070c799ebc8e5ce3dfe33d41b1838612ebda0e34362e90fR100), it should be `machine.openshift.io/v1beta1`:

`s/awsproviderconfig.openshift.io/v1beta1/machine.openshift.io/v1beta1/g`

File: modules/machineset-yaml-aws.adoc
Comment (yunjiang29) [inline modules/machineset-yaml-aws.adoc:129]: To be clear, this should be Local Zone public id created by `Creating a subnet in AWS Local Zones`, not the public id created by installer or VPC CF templates

**Review by yunjiang29 — COMMENTED** [review]
@dfitzmau thanks for the updates, the  general process looks good to me. I also have some comments here.

Other comments that I'm not sure where can I add review comments:

1. in https://65877--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating-infrastructure-machinesets#machineset-yaml-aws_creating-infrastructure-machinesets
  * Replace `awsproviderconfig.openshift.io/v1beta1` with `machine.openshift.io/v1beta1`
  * Remove `machine.openshift.io/cluster-api-machineset: <infrastructure_id>-edge-<localzone_az_name>`, the key already exists which value is `<infrastructure_id>-infra-<zone>`

2. in https://65877--docspreview.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-aws#machineset-yaml-aws_creating-machineset-aws
  * Remove `machine.openshift.io/cluster-api-machineset: <infrastructure_id>-edge-<localzone_az_name>`, the key already exists which value is `<infrastructure_id>-infra-<zone>`

File: modules/machineset-yaml-aws.adoc
Comment (mtulio) [reply modules/machineset-yaml-aws.adoc:129]: +1. maybe mentioning something like "Step 4" from the Section `Creating a subnet in AWS Local Zones`.
I just realized that there is no item in the legend for this field, it would be nice to add it there. @dfitzmau

File: modules/installation-aws-add-local-zone-locations.adoc
Comment (dfitzmau) [reply modules/installation-aws-add-local-zone-locations.adoc:60]: Updated.

Comment (dfitzmau) [reply modules/installation-aws-add-local-zone-locations.adoc]: Hi @mtulio . I checked the AWS docs and I see (https://aws.amazon.com/about-aws/global-infrastructure/localzones/locations/):

Screenshot from 2023-10-23 12-55-05

Comment (dfitzmau) [reply modules/installation-aws-add-local-zone-locations.adoc]: US EAST New York > US EAST (N.Virginia)?

File: modules/post-install-edge-aws-extend-cluster.adoc
Comment (adellape) [inline modules/post-install-edge-aws-extend-cluster.adoc:13]: Seems like we're split on this across the repo (and even within this assembly), but it doesn't seem like "Maximum Transmission Unit" needs to be init-capped (a general networking term). Unless this was maybe referring to a specific UI label somewhere.

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (adellape) [inline modules/post-install-edge-aws-extend-machineset.adoc:15]: Not going to block merge on this, but typically we would force-lowercase any user-replaceable values (so, these, and any other similar instances throughout the PR, like `<value_of_GroupName>`), though I realize they're likely referring to the literal capitalization of the related fields.

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (adellape) [inline modules/post-install-edge-aws-extend-machineset.adoc:25]: Extra space:
```suggestion
* Manually preserve `edge` machine pool labels when creating the machine set manifest by gathering the AWS API. To complete this action, enter the following command in your command-line interface (CLI):
```

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (adellape) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:61]: Missing `#` anchor.

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (adellape) [inline post_installation_configuration/aws-compute-edge-tasks.adoc:69]: Missing `#` anchor.

**Review by adellape — COMMENTED** [review]
I've added some nits that I wasn't going to block merge on, but saw the missing `#` anchors in the `xref`s near the end, and we should fix those up before merge. If you'd like to address any of the less-pondery other comments, feel free, but either way please re-queue for `merge-review-needed` when ready. Thanks!

File: modules/post-install-edge-aws-extend-machineset.adoc
Comment (dfitzmau) [reply modules/post-install-edge-aws-extend-machineset.adoc:15]: Thanks. That is correct given the nature of the literals and trying to be more direct so no confusion arises. Sometimes customer feedback comes in the form of changing these values , so using literals avoids this confusion.

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (dfitzmau) [reply post_installation_configuration/aws-compute-edge-tasks.adoc]: Well spotted. Updated.

File: modules/post-install-edge-aws-extend-cluster.adoc
Comment (dfitzmau) [reply modules/post-install-edge-aws-extend-cluster.adoc]: Updated numerous entities.

File: post_installation_configuration/aws-compute-edge-tasks.adoc
Comment (dfitzmau) [reply post_installation_configuration/aws-compute-edge-tasks.adoc]: Updated

**Review by adellape — APPROVED** [review]

**Pull request merged**
